projects/components/src/utils/test/angular-widget-driver.ts

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { DebugElement } from '@angular/core';
+import { ComponentFixture } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { AngularWidgetFinder } from './angular-widget-finder';
+import { WidgetDriver } from './base-widget-object';
+import { FindableWidget, FindParams } from './widget-object';
+
+/**
+ * A WidgetDriver that knows how to interface with the Angular framework.
+ */
+export class AngularWidgetDriver<T> implements WidgetDriver {
+    constructor(
+        protected fixture: ComponentFixture<any>,
+        protected root: DebugElement = fixture.debugElement,
+        protected component: T = fixture.componentInstance
+    ) {}
+
+    find<W extends WidgetDriver>(params: FindParams<FindableWidget<W>> | FindableWidget<W>): W {
+        return new AngularWidgetFinder(this.fixture).find<W, FindableWidget<W>>(params, this.root);
+    }
+
+    click(cssSelector?: string): Promise<void> {
+        this.getDebugElement(cssSelector).nativeElement.click();
+        this.fixture.detectChanges();
+        return Promise.resolve();
+    }
+
+    classes(cssSelector?: string): Promise<string[]> {
+        return Promise.resolve(Object.keys(this.getDebugElement(cssSelector).nativeElement.classes));
+    }
+
+    check(cssSelector?: string): Promise<void> {
```

Comment:
Mark your functions as `async` and you don't need to return  a Promise. That should clean the code a lot.